src/crawlers/browser_crawler.js

```diff
@@ -360,8 +360,43 @@ export default class BrowserCrawler extends BasicCrawler {
      * @internal
      */
     async _handleRequestFunction(crawlingContext) {
-        const { id } = crawlingContext;
-        const page = await this.browserPool.newPage({ id });
+        const newPageOptions = {
+            id: crawlingContext.id,
+        };
+
+        if (this.proxyConfiguration && this.launchContext.useIncognitoPages) {
+            const pageOptions = {};
+
+            const { session } = crawlingContext;
+            const { url } = this.proxyConfiguration.newProxyInfo(session && session.id);
+
+            const parsed = new URL(url);
+
+            // Playwright
+            pageOptions.proxy = {
+                server: parsed.origin,
+                username: parsed.username,
+                password: parsed.password,
+            };
+
+            // Puppeteer
+            pageOptions.proxyServer = parsed.origin;
+            pageOptions.proxyUsername = parsed.username;
+            pageOptions.proxyPassword = parsed.password;
+
+            // Disable SSL verification for MITM proxies
+            if (this.proxyConfiguration.isManInTheMiddle) {
+                /**
+                 * @see https://playwright.dev/docs/api/class-browser/#browser-new-context
+                 * @see https://github.com/puppeteer/puppeteer/blob/main/docs/api.md
+                 */
+                pageOptions.ignoreHTTPSErrors = true;
+            }
+
+            newPageOptions.pageOptions = pageOptions;
```

Comment:
Hmm, looking at this API I'm wondering if we should've done it as `{ id, ...pageOptions }` in `BrowserPool`.

Reply:
I was thinking the same, but `id` isn't a puppeteer / playwright option. Maybe `newPage(id, options)`? Do we need `id` at all?

In terms of `browser-pool` I think we should ignore `browserPlugin` since the crawler can be only Puppeteer or only Playwright. I think `browser-pool` should accept only one driver by design (currently it accepts many). The code-once run-anything idea unfortunately isn't true, since there will be always differences in API between Puppeteer and Playwright.

Reply:
> Maybe newPage(id, options)?
I was wondering the same, but then I wasn't sure whether we won't need more args than just `id` in the future.

> Do we need id at all?

Yeah, `id` is unfortunately needed to "link" a `crawlingContext` (or anything really) to a particular function invocation in the browser pool hooks e.g. [like here](https://github.com/apify/apify-js/blob/master/MIGRATIONS.md#configuration-and-lifecycle-hooks).

> I think browser-pool should accept...

Yeah, quite possibly. Please create an issue there not to derail the discussion here. Thanks!